filelog.go

```diff
@@ -251,9 +276,75 @@ func NewFileLogWriter(fname string, rotate bool) *FileLogWriter {
 		}
 	}()
 
+	// Background tasks goroutine
+	w.wg.Add(1)
+	go func() {
+		defer w.wg.Done()
+
+		for filename := range w.backgroundTasks {
+			if w.filesToKeep > 0 {
+				dir := filepath.Dir(filename)
+				err := w.archiveFiles(dir)
+				if err != nil {
+					fmt.Fprintf(w.errorWriter, "FileLogWriter(%q): Couldn't archive files: %s\n", filename, err)
+				}
+			}
+		}
+	}()
+
 	return w
 }
 
+func (w *FileLogWriter) archiveFiles(dir string) error {
+
+	// Get a handle to the directory
+	dirFile, err := os.Open(dir)
+	if err != nil {
+		return err
+	}
+
+	dirInfo, err := dirFile.Stat()
+	if err != nil {
+		return err
+	}
+	if !dirInfo.IsDir() {
+		return fmt.Errorf("%q must be a directory", dir)
+	}
+
+	var filesInDir []string
+	var matchedFiles []string
+
+	for {
+		filesInDir, err = dirFile.Readdirnames(1000)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		for _, fullFilename := range filesInDir {
+			baseFilename := filepath.Base(fullFilename)
+			if !w.logfileMatcher.MatchString(baseFilename) {
+				// Not interested in this file
+				continue
+			}
+
+			matchedFiles = append(matchedFiles, fullFilename)
+		}
+	}
+
+	sort.Strings(matchedFiles)
```

Comment:
Talked about this offline and we're planning to rip out the feature that creates sub-day rotations (e.g. `2017-01-01.001`) because we don't need it.